### PR TITLE
Cover missing cases for colon in WhiteSpaceRule.kt

### LINKED ISSUE #207: Colon with treeParent.elementType=FUN, not handled by WhiteSpaceRule
Follow-up on #151.
```
[INFO]    [ktlint] [WARN ] 2020-08-30 13:43:00 Colon with treeParent.elementType=FUN, not handled by WhiteSpaceRule
```
in current build log on project `diktat-common`.
Need to be investigated ad added to `WhiteSpaceRule`.

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Fixed logic, added more tests
* Edited code style guide

This pull request closes #207

### CODE REVIEW

**Review by kentr0w — APPROVED** [review]
LGTM

File: info/diktat-kotlin-coding-style-guide-en.md [hunk 1]
```diff
@@ -1092,6 +1092,7 @@ fun baz() {
   There should be no spaces before `,`, `:` and `;`. The only exceptions for colon are the following:
   - when `:` is used to separate a type and a supertype, including anonimous object (after `object` keyword)
   - when delegating to a superclass constructor or a different constructor of the same class
+  - when colon is used in annotation to specify use-site target (e.g. `@param:JsonProperty`)
```
Comment (akuleshov7) [inline info/diktat-kotlin-coding-style-guide-en.md:1095]: We are saying that here should be a space? But in example (in e.g) we are saying that here should be no spaces...

**Review by akuleshov7 — CHANGES_REQUESTED** [review]

File: info/diktat-kotlin-coding-style-guide-en.md (hunk 1 above)
Comment (petertrr) [reply info/diktat-kotlin-coding-style-guide-en.md]: Fair enough, changed.

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**